spi/src/main/java/com/flipkart/security/grayskull/spi/AuthenticationHandler.java

```diff
@@ -0,0 +1,12 @@
+package com.flipkart.security.grayskull.spi;
+
+import jakarta.servlet.http.HttpServletRequest;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.core.Authentication;
+
+public interface AuthenticationHandler {
```

Comment:
spring boot does not use "handler" as a keyword. we can decide what it is actually. most likely we would be implementing this as filter.